roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py

```diff
@@ -51,26 +51,17 @@ def __init__(self, file_path, mov_field='mov', sampling_frequency=None,
         self._kwargs = {'file_path': str(Path(file_path).absolute()), 'mov_field': mov_field,
                         'sampling_frequency': sampling_frequency, 'channel_names': channel_names}
 
-    def get_frame(self, frame_idx, channel=0):
-        assert frame_idx < self.get_num_frames()
-        return self._video[channel, frame_idx]
-
+    @check_get_frames_args
     def get_frames(self, frame_idxs, channel=0):
-        frame_idxs = np.array(frame_idxs)
-        assert np.all(frame_idxs < self.get_num_frames())
-        sorted_frame_idxs, sorting_inverse = np.sort(frame_idxs, return_inverse=True)
-        return self._video[channel, sorted_frame_idxs][:, sorting_inverse]
+        if frame_idxs.size > 1 and np.all(np.diff(frame_idxs) > 0):
+            return self._video[channel, frame_idxs]
+        else:
+            sorted_frame_idxs, sorting_inverse = np.sort(frame_idxs, return_inverse=True)
+            return self._video[channel, sorted_frame_idxs][:, sorting_inverse]
```

Comment:
Do we want this to continue to be a memmap for HDF5? `lazy_ops` can do that with:

`lazy_ops.DatasetView(self._video).lazy_slice[channel, frame_idxs]`

and 

`lazy_ops.DatasetView(self._video).lazy_slice[channel, sorted_frame_idxs].lazy_slice[:, sorting_inverse]`

Reply:
Sure let's do that!